bot/cogs/embeds.py

```diff
@@ -0,0 +1,327 @@
+
+        if embed is False:
+            return
+
+        embed.colour = color
+        self.embed[ctx.author] = embed
+        await ctx.send("Embeds color updated")
+
+    # region: author
+    @embed_group.command(name="author", aliases=["setauthor", "authorname"])
+    @with_role(*MODERATION_ROLES)
+    async def embed_author_name(self, ctx: Context, *, author_name: str) -> None:
+        """Set authors name in embed"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        embed.set_author(name=author_name, url=embed.author.url,
+                         icon_url=embed.author.icon_url)
+        self.embed[ctx.author] = embed
+        await ctx.send("Embeds author updated")
+
+    @embed_group.command(name="authorurl", aliases=["setauthorurl"])
+    @with_role(*MODERATION_ROLES)
+    async def embed_author_url(self, ctx: Context, *, author_url: str) -> None:
+        """Set authors URL in embed"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        embed.set_author(name=embed.author.name, url=author_url,
+                         icon_url=embed.author.icon_url)
+        self.embed[ctx.author] = embed
+        await ctx.send("Embeds author URL updated")
+
+    @embed_group.command(name="authoricon", aliases=[
+        "setauthoricon", "authoriconurl", "setauthoriconurl"
+    ])
+    @with_role(*MODERATION_ROLES)
+    async def embed_author_icon(self, ctx: Context, *, icon_url: str) -> None:
+        """Set authors image in embed"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        embed.set_author(name=embed.author.name, url=embed.author.url,
+                         icon_url=icon_url)
+        self.embed[ctx.author] = embed
+        await ctx.send("Embeds authors image updated")
+    # endregion
+
+    # region: fields
+    @embed_group.command(name="createfield", aliases=[
+        "newfield", "makefield", "addfield",
+        "fieldadd", "fieldcreate", "fieldmake", "field"
+    ])
+    @with_role(*MODERATION_ROLES)
+    async def embed_field_create(self, ctx: Context, *, title: str = "None") -> None:
+        """Create new field in embed"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        embed.add_field(name=title, value="None")
+
+        self.embed_field_id[ctx.author] += 1
+        self.embed[ctx.author] = embed
+        await ctx.send(f"Embed field with ID **{self.embed_field_id[ctx.author]}** created")
+
+    @embed_group.command(name="fielddescription", aliases=["fieldvalue"])
+    @with_role(*MODERATION_ROLES)
+    async def embed_field_description(self, ctx: Context, ID: int, *, description: str) -> None:
+        """Set description of embeds field"""
+        embed = await self.get_embed(ctx)
+
+        if embed is False:
+            return
+
+        if self.embed_field_id[ctx.author] < ID:
```

Comment:
Prevent users from using negative IDs.
```suggestion
        if self.embed_field_id[ctx.author] < ID and ID >= 0:
```